pkg/config/parameters.go

```diff
@@ -413,6 +413,13 @@ type ClusterParameters struct {
 	// See https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto.html#envoy-v3-api-enum-config-cluster-v3-cluster-dnslookupfamily
 	// for more information.
 	DNSLookupFamily ClusterDNSFamilyType `yaml:"dns-lookup-family"`
+
+	// Defines the maximum requests for upstream connections. If not specified, there is no limit.
+	// see https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/protocol.proto#envoy-v3-api-msg-config-core-v3-httpprotocoloptions
+	// for more information.
+	//
+	// +optional
+	MaxRequestsPerConnection *uint32 `yaml:"max-requests-per-connection,omitempty"`
```

Comment:
sorry for the runaround but a similar minimum value of 1 validation in this package would be good to add

Reply:
ah i wasn't clear here, but we'll need to add validation in the go code itself, this is not a CRD so kubebuilder validations don't apply, we'll need to add something here https://github.com/projectcontour/contour/blob/4333ed3bbab811a4517854f2d892028f2e268df8/pkg/config/parameters.go#L825 and a test to verify it

Reply:
oops, sorry for the back and forth. hopefully this last commit should take care of it.